OSD code doesn't use new event system!

Breaks build.
```
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c: In function 'OnScreenDisplayInitialize':
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1623:41: warning: passing argument 1 of 'OnScreenDisplaySettingsConnectCallback' from incompatible pointer type
  OnScreenDisplaySettingsConnectCallback(OnScreenSettingsUpdatedCb);
                                         ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:66:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaysettings.h:93:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplaySettingsConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplaySettingsHandle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1626:45: warning: passing argument 1 of 'OnScreenDisplayPageSettingsConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettingsConnectCallback(OnScreenPageUpdatedCb);
                                             ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:67:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettingsConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettingsHandle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1627:46: warning: passing argument 1 of 'OnScreenDisplayPageSettings2ConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettings2ConnectCallback(OnScreenPageUpdatedCb);
                                              ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:68:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings2.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettings2ConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettings2Handle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1628:46: warning: passing argument 1 of 'OnScreenDisplayPageSettings3ConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettings3ConnectCallback(OnScreenPageUpdatedCb);
                                              ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:69:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings3.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettings3ConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettings3Handle(), cb, NULL, EV_MASK_ALL_UPDATES); }
                       ^
/home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:1629:46: warning: passing argument 1 of 'OnScreenDisplayPageSettings4ConnectCallback' from incompatible pointer type
  OnScreenDisplayPageSettings4ConnectCallback(OnScreenPageUpdatedCb);
                                              ^
In file included from /home/mike/Dev/dronin/flight/Modules/OnScreenDisplay/onscreendisplay.c:70:0:
/home/mike/Dev/dronin/build/uavobject-synthetics/flight/onscreendisplaypagesettings4.h:209:23: note: expected 'UAVObjEventCallback' but argument is of type 'void (*)(struct UAVObjEvent *)'
 static inline int32_t OnScreenDisplayPageSettings4ConnectCallback(UAVObjEventCallback cb) { return UAVObjConnectCallback(OnScreenDisplayPageSettings4Handle(), cb, NULL, EV_MASK_ALL_UPDATES); }
```

--- a/flight/Modules/OnScreenDisplay/onscreendisplay.c
+++ b/flight/Modules/OnScreenDisplay/onscreendisplay.c
@@ -1550,18 +1550,6 @@ int render_stats()
 }
 
 
-static void OnScreenSettingsUpdatedCb(UAVObjEvent * ev)
-{
-	osd_settings_updated = true;
-}
-
-
-static void OnScreenPageUpdatedCb(UAVObjEvent * ev)
-{
-	osd_page_updated = true;
-}
-
-
 /**
  * Start the osd module
  */
@@ -1620,13 +1608,13 @@ int32_t OnScreenDisplayInitialize(void)
 	OnScreenDisplayPageSettings4Initialize();
 
 	/* Register callbacks for modified settings */
-	OnScreenDisplaySettingsConnectCallback(OnScreenSettingsUpdatedCb);
+	OnScreenDisplaySettingsConnectCallbackCtx(UAVObjCbSetFlag, &osd_settings_updated);
 
 	/* Register callbacks for modified pages */
-	OnScreenDisplayPageSettingsConnectCallback(OnScreenPageUpdatedCb);
-	OnScreenDisplayPageSettings2ConnectCallback(OnScreenPageUpdatedCb);
-	OnScreenDisplayPageSettings3ConnectCallback(OnScreenPageUpdatedCb);
-	OnScreenDisplayPageSettings4ConnectCallback(OnScreenPageUpdatedCb);
+	OnScreenDisplayPageSettingsConnectCallbackCtx(UAVObjCbSetFlag, &osd_page_updated);
+	OnScreenDisplayPageSettings2ConnectCallbackCtx(UAVObjCbSetFlag, &osd_page_updated);
+	OnScreenDisplayPageSettings3ConnectCallbackCtx(UAVObjCbSetFlag, &osd_page_updated);
+	OnScreenDisplayPageSettings4ConnectCallbackCtx(UAVObjCbSetFlag, &osd_page_updated);
 
 	return 0;
 }

